### PR TITLE
Feature/dsp 349 retrieve json column

### PR DESCRIPTION
**Pull request by zhangchi1**
**What this PR does / why we need it**:
Update snowflake retriever with its SQL template, in order to adapt **variant** column
**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: storage/connectors/jdbc/src/test/java/feast/storage/connectors/jdbc/retriever/SnowflakeHistoricalRetrieverJSONColTest.java [hunk 1]
```diff
@@ -23,12 +23,15 @@
 import feast.storage.api.retriever.HistoricalRetrievalResult;
 import feast.storage.connectors.jdbc.connection.SnowflakeConnectionProvider;
 import java.io.File;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
```
Comment (zhangchi1) [inline storage/connectors/jdbc/src/test/java/feast/storage/connectors/jdbc/retriever/SnowflakeHistoricalRetrieverJSONColTest.java:29]: it seems like `mvn spotless:apply` automatically cleaned the format this time?

File: storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/retriever/AbstractJdbcQueryTemplater.java [hunk 2]
```diff
@@ -254,7 +254,8 @@ protected String createTimestampLimitQuery(String leftTableName) {
       for (String entity : featureSetQueryInfo.getEntities()) {
         if (!entities.contains(entity)) {
           entities.add(entity);
-          entityColumns.add(String.format("%s.%s", table, entity));
+          // parse entities from FEATURE variant column
+          entityColumns.add(String.format("%s.feature:%s AS %s", table, entity, entity));
```
Comment (jmelinav) [inline storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/retriever/AbstractJdbcQueryTemplater.java]: is this specific to snowflake?

Comment (zhangchi1) [reply storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/retriever/AbstractJdbcQueryTemplater.java]: good point. let me double check

**Review by dr3s — CHANGES_REQUESTED** [review]
just fix the review comments.  primarily the one from @jmelinav

File: storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/retriever/AbstractJdbcQueryTemplater.java (hunk 2 above)
Comment (zhangchi1) [reply storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/retriever/AbstractJdbcQueryTemplater.java]: @dr3s @jmelinav I have a question regarding the Postgres implementation. In this PR, shall I also update the Postgres templater? Or, we don't support Postgres for now?

**Review by dr3s — APPROVED** [review]

File: storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/retriever/AbstractJdbcQueryTemplater.java (hunk 2 above)
Comment (swatiarora0208) [reply storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/retriever/AbstractJdbcQueryTemplater.java]: @zhangchi1 for now you can comment the postgres implementation and make sure to move snowflake specific code under snowflake classes.

**Pull request merged**